DeusEx/Classes/DXRReduceItems.uc

```diff
@@ -1,20 +1,119 @@
 class DXRReduceItems extends DXRActorsBase;
 
+struct sReduceAmmo {
+    var string type;
+    var int percent;
+};
+struct sReduceItem {
+    var string type;
+    var int percent;
+};
+struct sSetMax {
+    var string type;
+    var int percent;
+};
+struct sMaxAmmo {
+    var string type;
+    var int percent;
+};
+
+var config int mission_scaling[16];
+var config sReduceAmmo ammo_reductions[16];
+var config sReduceItem reduce_items[16];
+var config sSetMax max_copies[16];
+var config sMaxAmmo max_ammo[16];
+
+function CheckConfig()
+{
+    local int i;
+    if( config_version == 0 ) {
+        for(i=0; i < ArrayCount(mission_scaling); i++) {
+            mission_scaling[i] = 100;
+        }
+        for(i=0; i < ArrayCount(ammo_reductions); i++) {
+            ammo_reductions[i].type = "";
+        }
+        for(i=0; i < ArrayCount(reduce_items); i++) {
+            reduce_items[i].type = "";
+        }
+        for(i=0; i < ArrayCount(max_copies); i++) {
+            max_copies[i].type = "";
+        }
+        for(i=0; i < ArrayCount(max_ammo); i++) {
+            max_ammo[i].type = "";
+        }
+    }
+    Super.CheckConfig();
+}
+
 function FirstEntry()
 {
+    local int i, mission, scale;
+    local class<Actor> c;
     Super.FirstEntry();
 
-    ReduceAmmo( float(dxr.flags.ammo)/100.0 );
-    ReduceSpawns(class'Multitool', dxr.flags.multitools);
-    ReduceSpawns(class'Lockpick', dxr.flags.lockpicks);
-    ReduceSpawns(class'BioelectricCell', dxr.flags.biocells);
-    ReduceSpawns(class'MedKit', dxr.flags.medkits);
+    mission = Clamp(dxr.dxInfo.missionNumber, 0, ArrayCount(mission_scaling)-1);
+    scale = mission_scaling[mission];
+
+    ReduceAmmo(class'Ammo', float(dxr.flags.ammo*scale)/100.0/100.0);
+
+    ReduceSpawns(class'Multitool', dxr.flags.multitools*scale/100);
+    ReduceSpawns(class'Lockpick', dxr.flags.lockpicks*scale/100);
+    ReduceSpawns(class'BioelectricCell', dxr.flags.biocells*scale/100);
+    ReduceSpawns(class'MedKit', dxr.flags.medkits*scale/100);
+
+    for(i=0; i < ArrayCount(ammo_reductions); i++) {
+        if( ammo_reductions[i].type == "" ) continue;
+        c = GetClassFromString( ammo_reductions[i].type, class'Ammo' );
+        ReduceAmmo(class<Ammo>(c), float(ammo_reductions[i].percent*scale)/100.0/100.0 );
+    }
+    for(i=0; i < ArrayCount(reduce_items); i++) {
+        if( reduce_items[i].type == "" ) continue;
+        c = GetClassFromString( reduce_items[i].type, class'Actor' );
+        ReduceSpawns(c, reduce_items[i].percent*scale/100 );
+    }
+    SetAllMaxCopies(scale);
```

Comment:
maybe we should have a config option to scale MaxCopies differently from the spawn rate, I would like to see maybe `SetAllMaxCopies(scale * 0.8);` or something